src/shared/components/case-editor/case-edit-page/case-edit-page.component.ts

```diff
@@ -127,8 +127,11 @@ export class CaseEditPageComponent implements OnInit, AfterViewChecked {
       let caseEventData: CaseEventData = this.formValueService.sanitise(pageFormFields) as CaseEventData;
       caseEventData.event_token = this.eventTrigger.event_token;
       caseEventData.ignore_warning = this.ignoreWarning;
-      caseEventData.event_data = this.editForm.value.data;
-
+      let event_data = this.editForm.value.data
+      Object.keys(pageFormFields.data).forEach((key) => {
+        event_data[key] = pageFormFields.data[key];
```

Comment:
@mohammedanas one thing. I'm not sure but can you please verify the following: 
You are reading the fields from `pageFormFields`. But `pageFormFields` only has got the current page fields. We need to make sure 
also the dynamic fields in the previous pages are passed correctly to the MidEvent callback. Can you verify that happens please?

Reply:
Already thought about it Mario, So as per code its not going to update the previous fields so they remain same.. we deal on per page basis, thats fine as well

Reply:
would it be possible to add a quick unit test to confirm that and avoid regressions? if this doesn't take too much time

Reply:
I think, It will be more of setting up scenarios of multiple pages and interactions between them, not sure how we can acheive them, in a unit test. I am open to options, but at this stage and urgency, manual tests are fine.